internal/pkg/queue/stats.go

```diff
@@ -174,9 +174,6 @@ func (q *PersistentGroupedQueue) calculateAverageDuration(samples []Sample) OpAv
 }
 
 func (q *PersistentGroupedQueue) addSample(duration time.Duration, sampleType SampleType) {
-	q.stats.Lock()
```

Comment:
you sure this will not cause any data race?

Reply:
> you sure this will not cause any data race?

I'm not sure any lock is needed at the stats level here considering the entire queue struct (which hold the stats) is locked in Enqueue and Dequeue ?